### PR TITLE
add prune test

### PR DESCRIPTION
**Pull request by whitewindmills**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
add prune test
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/resourceinterpreter/defaultinterpreter/prune/prune_test.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+	"github.com/karmada-io/karmada/pkg/util"
+)
+
+type field []string
+
+func (f field) String() string {
+	return "." + strings.Join(f, ".")
+}
+
+type Test struct {
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/defaultinterpreter/prune/prune_test.go:19]: This structure seems does not need to be exported.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks a lot~

File: pkg/resourceinterpreter/defaultinterpreter/prune/prune_test.go (hunk 1 above)
Comment (whitewindmills) [reply pkg/resourceinterpreter/defaultinterpreter/prune/prune_test.go:19]: > This structure seems does not need to be exported.

Thanks for your suggestion, I ignored this.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

File: pkg/resourceinterpreter/defaultinterpreter/prune/prune_test.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+		{
+			name: "remove serviceaccount irrelevant fields",
+			workload: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind": util.ServiceAccountKind,
+					"metadata": map[string]interface{}{
+						"name": "foo",
+					},
+					"secrets": []interface{}{
+						conv(map[string]interface{}{
```
[the remark this answers is not in the document]
Comment (whitewindmills) [reply pkg/resourceinterpreter/defaultinterpreter/prune/prune_test.go:143]: I don't think so, you can remove `conv` function and rerun this unit test.

[the remark this answers is not in the document]
Comment (whitewindmills) [reply pkg/resourceinterpreter/defaultinterpreter/prune/prune_test.go:143]: Sorry, this is a function I forgot to remove while debugging.

**Pull request merged**